### PR TITLE
Fix permission requests on upgrading to the one-way video

### PR DESCRIPTION
**Pull request by AndriiHorishniiMOC**
Prevent permission requests when upgrading to the one-way video

[MOB-1459](https://glia.atlassian.net/browse/MOB-1459)

### CODE REVIEW

**Review by DavDo — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java [hunk 1]
```diff
@@ -143,12 +143,14 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
...
-                && missingPermission(Manifest.permission.RECORD_AUDIO)) {
-            missingPermissions.add(Manifest.permission.RECORD_AUDIO);
+        if (!configuration.getIsUpgradeToCall()) {
+            if (configuration.getMediaType() == Engagement.MediaType.VIDEO && missingPermission(Manifest.permission.CAMERA)) {
+                missingPermissions.add(Manifest.permission.CAMERA);
+            }
+            if ((configuration.getMediaType() == Engagement.MediaType.VIDEO || configuration.getMediaType() == Engagement.MediaType.AUDIO)
+                    && missingPermission(Manifest.permission.RECORD_AUDIO)) {
+                missingPermissions.add(Manifest.permission.RECORD_AUDIO);
+            }
```
Comment (andrews-moc) [inline widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:153]: Looking at this code, I would expect that:
 `if need to upgrade to call` => then `need to ask permissions`.

But it's strange to me that it's vice-a-versa: `!configuration.getIsUpgradeToCall()`. Could you please explain the logic?

Comment (AndriiHorishniiMOC) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:153]: Core SDK can request permissions by itself. So upgrading to call handles on the Core SDK side.
As far as I understand, those permission requests from here are a way to prepare all needed for the call screen before making a deal with the Core SDK.

Comment (andrews-moc) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:153]: Ok, maybe it makes sense to add some comment here.

**Review by andrews-moc — APPROVED** [review]

File: widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java (hunk 1 above)
Comment (gugalo) [reply widgetssdk/src/main/java/com/glia/widgets/call/CallActivity.java:153]: > As far as I understand, those permission requests from here are a way to prepare all needed for the call screen before making a deal with the Core SDK.  

Yes. There was some scenario in Widget SDK that caused an expected behavior. But I don't remember why and can't find the bug ticket

**Comment by AndriiHorishniiMOC** [comment]
!squash

**Comment by AndriiHorishniiMOC** [comment]
!merge

**Pull request merged**